Expensify/App
# Add indicators to let the user know if they are on staging or dev

@roryabraham @shawnborton will you please review this?

### Details
This PR adds visuals to the sign in screen, the main app screen, and the desktop icon to let the user know what environment they are running the app in (dev/staging/prod). 

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/158445

### Tests
1. Run the app on dev on all platforms and sign out, confirm that you see a `dev` badge on the sign-in screen logo. On desktop, confirm you see a `dev` badge for the doc icon.
2. Login to any account, confirm after logging in you see a `dev` badge on the main app screen next to `Chats`, and that the badge also shows up in `Settings`
3. On desktop, run `npm run desktop-build-staging`, then open the Expensify.cash app inside `dist/mac`. Confirm the icon has the `stg` badge.
4. Confirm on the sign-in screen the icon has a `stg` badge. Sign in and confirm the `stg` badge shows up next to `Chats` and `Settings`
5. Update your `.env` file to include `ENVIRONMENT=STAGING` and run the app on web/ios/android, confirm you see the `stg` badge on the sign-in screen and in the app
6. Update your `.env` file to include `ENVIRONMENT=PRODUCTION` and run the app on web/ios/android, confirm you see no badges on the sign-in screen and in the app

### QA Steps
1. Confirm that when running the app on staging, a `STG` badge appears on the sign in screen icon, and in the main app next to `Chats` and `Settings` (after login)
    2. Confirm that on desktop, the desktop icon shows with the `STG` badge
2. Confirm that when running the app on prod, there are no badges

### Tested On


### Screenshots

#### Web/Mobile Web/Desktop
| Dev | Staging | Prod |
|---|---|---|
| ![image](https://user-images.githubusercontent.com/3981102/115637162-5e0afc00-a2c4-11eb-838c-bcaa4fe49937.png) | ![image](https://user-images.githubusercontent.com/3981102/115637171-62371980-a2c4-11eb-976b-83457c638131.png) | ![image](https://user-images.githubusercontent.com/3981102/115637181-66fbcd80-a2c4-11eb-8189-afb27e835f27.png) |
| ![image](https://user-images.githubusercontent.com/3981102/115637200-6f540880-a2c4-11eb-93f7-efe6160f6b0b.png) | ![image](https://user-images.githubusercontent.com/3981102/115637204-724ef900-a2c4-11eb-86bc-cc7f48d9bc63.png) | ![image](https://user-images.githubusercontent.com/3981102/115637393-eab5ba00-a2c4-11eb-8875-8d670e5528f8.png) |


#### iOS/Android
| Dev | Staging |
|---|---|
| ![image](https://user-images.githubusercontent.com/3981102/115637539-39fbea80-a2c5-11eb-9d59-c521879c0904.png) | ![image](https://user-images.githubusercontent.com/3981102/115637430-fb663000-a2c4-11eb-8609-f2dd15f07bb9.png) |
| ![image](https://user-images.githubusercontent.com/3981102/115637525-336d7300-a2c5-11eb-9391-3786dbca0b4d.png) | ![image](https://user-images.githubusercontent.com/3981102/115637438-00c37a80-a2c5-11eb-9dea-246222bb06ea.png) |
